summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -194,59 +194,52 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
       }
     }
 
-    val countersForSpout: Seq[(Group, Name)] = JobCounters.getCountersForJob(jobID).getOrElse(Nil)
     val isMergeableWithSource = getOrElse(stormDag, node, DEFAULT_FM_MERGEABLE_WITH_SOURCE).get
     val sourceParallelism = getOrElse(stormDag, node, parOpt.getOrElse(DEFAULT_SOURCE_PARALLELISM)).parHint
     val flatMapParallelism = getOrElse(stormDag, node, DEFAULT_FM_PARALLELISM).parHint
 
     if (isMergeableWithSource) {
       require(flatMapParallelism <= sourceParallelism, s"SourceParallelism ($sourceParallelism) must be at least as high as FlatMapParallelism ($flatMapParallelism) when merging flatMap with Source")
     }
-    val metrics = getOrElse(stormDag, node, DEFAULT_SPOUT_STORM_METRICS)
 
+    val summerOpt: Option[SummerNode[Storm]] = stormDag.dependantsOf(node.asInstanceOf[StormNode]).collect { case s: SummerNode[Storm] => s }.headOption
+    val builder = if (summerOpt.isDefined) BuildSummer(this, stormDag, node, jobID)
+    val metrics = getOrElse(stormDag, node, DEFAULT_SPOUT_STORM_METRICS)
+    val countersForSpout: Seq[(Group, Name)] = JobCounters.getCountersForJob(jobID).getOrElse(Nil)
     val registerAllMetrics = Externalizer({ context: TopologyContext =>
+
       // Register metrics passed in SpoutStormMetrics option.
       metrics.metrics().foreach {
         x: StormMetric[IMetric] =>
           context.registerMetric(x.name, x.metric, x.interval.inSeconds)
       }
       // Register summingbird counter metrics.
+      println("registerAllMetrics call jobId is %s\n countersForSpout is : %s".format(jobID.toString, countersForSpout.toString()))
       StormStatProvider.registerMetrics(jobID, context, countersForSpout)
       SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
     })
-    val hookedTormentaSpout = tormentaSpout.openHook(registerAllMetrics.get)
-
-    val summerOpt: Option[SummerNode[Storm]] = stormDag.dependantsOf(node.asInstanceOf[StormNode]).collect { case s: SummerNode[Storm] => s }.headOption
 
     val stormSpout = summerOpt match {
-      /*
-         *  As the spout is being followed by a summer, we do not have map-side aggregation handled in the spout.
-         *  This means Summer is the place where the aggregation should happen.
-         */
-      case Some(s) => createSpoutToFeedSummer(stormDag, s, hookedTormentaSpout)
-      case None => hookedTormentaSpout.getSpout
+      case Some(s) => createSpoutToFeedSummer[Any, Any](stormDag, s, builder.asInstanceOf[SummerBuilder], tormentaSpout, registerAllMetrics)
+      case None => tormentaSpout.openHook(registerAllMetrics.get).getSpout
```

Comment:
I think it makes sense to create a helper method that combines the `openHook` and `getSpout` calls.  Something like:
```scala
// Tormenta bug where openHook might get lost if it doesn't happen right before getSpout
def getStormSpout[T](spout: Spout[T], jobId: JobId): IRichSpout = {
  val countersForSpout = ... // use jobId
  val hookedSpout = spout.openHook(...)
  hookedSpout.getSpout
}
```
This will dedupe these calls while logically grouping them so it's harder to do the wrong thing (e.g., calling `getCountersForJob` or `openHook` too early).